### PR TITLE
GEOPY-2714: Refactor mask_by_extent utilities for surfaces

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-2714 - Refactor mask_by_extent utilities for surfaces**

### CODE REVIEW

File: geoh5py/objects/grid_object.py [hunk 1]
```diff
@@ -61,6 +62,17 @@ def centroids(self) -> np.ndarray:
...
+    @property
+    def extent(self) -> np.ndarray:
+        """
+        Compute outer extent of mesh span in world coordinates.
+        """
+        u, v, w = np.meshgrid(
+            np.r_[0, self.span[0]], np.r_[0, self.span[1]], np.r_[0, self.span[2]]
+        )
+        xyz = self.uvw_to_xyz(np.c_[u.ravel(), v.ravel(), w.ravel()])
+        return np.c_[xyz.min(axis=0), xyz.max(axis=0)].T
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/grid_object.py:75]: Same comment as the following one - done.

File: geoh5py/objects/octree.py [hunk 2]
```diff
@@ -209,6 +209,19 @@ def shape(self) -> tuple[np.int32, np.int32, np.int32]:
...
         return self.u_count, self.v_count, self.w_count
 
+    @property
+    def span(self) -> np.ndarray:
+        """
+        Upper and lower limits along u, v and w directions.
+        """
+        return np.vstack(
+            [
+                np.sort([0, self.u_cell_size * self.u_count]),
```
Comment (MatthieuCMira) [inline geoh5py/objects/octree.py:219]: right, cell_size can be negative...

File: geoh5py/objects/vp_model.py [hunk 3]
```diff
@@ -395,6 +414,24 @@ def u_count(self) -> np.int32:
...
 
+    def uvw_to_xyz(self, coordinates: np.ndarray) -> np.ndarray:
+        """
+        Apply rotation to the input coordinates.
+
+        Deal with clockwise rotation angle of VPMesh
+
+        :param coordinates: Array of coordinates along the axes, transformed to world coordinates.
+
+        :return:
```
Comment (MatthieuCMira) [inline geoh5py/objects/vp_model.py]: return?

**Review by MatthieuCMira — DISMISSED** [review]
I agree with the change.

I have the feeling we should add test for negative cell size values.

File: geoh5py/objects/octree.py (hunk 2 above)
Comment (domfournier) [reply geoh5py/objects/octree.py:219]: Done

**Review by MatthieuCMira — APPROVED** [review]
seems good to me.
